main.js

```diff
@@ -1 +1,48 @@
-// write your code here to make the tests pass
+const Library = () => {
+  const books = [];
+  const isBookFound = (book) => books.findIndex(libraryBook => {
+    return libraryBook.getAttribute('title') === book.getAttribute('title');
+  }) > -1;
+  const errorMessage = (book) => `Book '${book.getAttribute('title')}' does not exist in the library`;
+
+  return {
+    addBook(book) {
+      books.push(book);
+    },
+    checkOutBook(book) {
+      if (isBookFound(book)) {
+        return book.setAttribute('checkedOut', true);
+      }
+
+      console.log(errorMessage(book));
```

Comment:
I prefer to handle the negative case first, and then let the positive case happen after that.  So handle the !isBookFound case then do the setAttribute call later. It keeps the check and the result closer together.

Reply:
Yes that makes sense. I'll refactor the code to fix this.